import_export/src/file/export.rs

```diff
@@ -0,0 +1,240 @@
+use futures_util::TryStreamExt;
+use influxdb_iox_client::{
+    catalog::{self, generated_types::ParquetFile},
+    connection::Connection,
+    store,
+};
+use observability_deps::tracing::{debug, info};
+use std::path::{Path, PathBuf};
+use thiserror::Error;
+use tokio::{
+    fs::{self, File, OpenOptions},
+    io::{self, AsyncWriteExt},
+};
+use tokio_util::compat::FuturesAsyncReadCompatExt;
+
+#[derive(Debug, Error)]
+pub enum ExportError {
+    #[error("JSON Serialization error: {0}")]
+    Serde(#[from] serde_json::Error),
+
+    #[error("IOx request failed: {0}")]
+    ClientError(#[from] influxdb_iox_client::error::Error),
+
+    #[error("Writing file: {0}")]
+    FileError(#[from] std::io::Error),
+}
+
+type Result<T, E = ExportError> = std::result::Result<T, E>;
+
+/// Exports data from a remote IOx instance to local files.
+///
+/// Data is read using the clients in [`influxdb_iox_client`] (rather
+/// than the catalog) so that this can be used to debug remote systems.
+#[derive(Debug)]
+pub struct RemoteExporter {
+    catalog_client: catalog::Client,
+    store_client: store::Client,
+
+    /// Optional partition filter. If `Some(partition_id)`, only these
+    /// files with that `partition_id` are downloaded.
+    partition_filter: Option<i64>,
+}
+
+impl RemoteExporter {
+    pub fn new(connection: Connection) -> Self {
+        Self {
+            catalog_client: catalog::Client::new(connection.clone()),
+            store_client: store::Client::new(connection),
+            partition_filter: None,
+        }
+    }
+
+    pub fn with_partition_filter(mut self, partition_filter: Option<i64>) -> Self {
+        if let Some(partition_filter) = partition_filter {
+            info!(partition_filter, "Filtering by partition");
+        }
+
+        self.partition_filter = partition_filter;
+        self
+    }
+
+    /// Exports all data and metadata for `table_name` in
+    /// `namespace` to local files.
+    ///
+    /// If `output_directory` is specified, all files are written
+    /// there otherwise files are exported to a directory named
+    /// `table_name`.
+    pub async fn export_table(
+        &mut self,
+        output_directory: Option<PathBuf>,
+        namespace_name: String,
+        table_name: String,
+    ) -> Result<()> {
+        let output_directory = output_directory.unwrap_or_else(|| PathBuf::from(&table_name));
+        fs::create_dir_all(&output_directory).await?;
+
+        let parquet_files = self
+            .catalog_client
+            .get_parquet_files_by_namespace_table(namespace_name, table_name)
+            .await?;
+
+        // try and export all relevant tables (all parquet files
+        // should share the same table_id (as they are all the same
+        // name) but there is no catalog gRPC call to get a table_id by name
```

Comment:
Nit: unclosed delimiter, brain compiling failed. Also I'm not sure what "try and export all relevant tables" means as there should only be one table? Should this say "try and export the relevant table"? Or "try and export all relevant parquet files?" I don't know :)

 ```suggestion
        // try and export all relevant tables (all parquet files
        // should share the same table_id as they are all the same
        // name) but there is no catalog gRPC call to get a table_id by name
```

Also also, "there is no catalog gRPC call to get a table_id by name"... if only we knew who implemented the catalog gRPC calls and could ask them to add a new one....? 🤔 😉

Reply:
I agree this comment is confusing. I reworded it to

```
        // Export the metadata for the table. Since all
        // parquet_files are part of the same table, use the table_id
        // from the first parquet_file
```

in a616835d3